index.js

```diff
@@ -24,14 +24,55 @@ function Plugin(inputNodes, options) {
     }
   }
 
+
+  this._lastInputTreeRevisions = inputNodes.map(function() { return NaN; });
+  this._revision = NaN;
+
   this._baseConstructorCalled = true
   this._inputNodes = inputNodes
   this._persistentOutput = !!options.persistentOutput
+  this._sideEffectFree = !!options.sideEffectFree
   this._needsCache = (options.needsCache != null) ? !!options.needsCache : true
 
   this._checkOverrides()
 }
 
+
+function getRevision(tree) {
```

Comment:
This might be simpler to reason about if it were inlined, we only use it in the one place...